python/svi_client/src/hydrotools/svi_client/clients.py

```diff
@@ -0,0 +1,220 @@
+            year=year,
+            geographic_context=geographic_context,
+            count_only=True,
+        )
+
+        # RestClient only allows 200 response code or an aiohttp.client_exceptions.ClientConnectorError is raised
+        # number of features
+        count_request = self._rest_client.get(url_path)
+
+        deserialized_count = count_request.json()
+        count = deserialized_count["properties"]["count"]
+
+        # number of features requested by a single request
+        OFFSET = 1000
+        n_gets = (count // OFFSET) + 1
+
+        urls = [
+            url_builders.build_feature_server_url(
+                location=location,
+                geographic_scale=geographic_scale,
+                year=year,
+                geographic_context=geographic_context,
+                result_offset=i * OFFSET,
+                result_record_count=OFFSET,
+            )
+            for i in range(n_gets)
+        ]
+
+        results = self._rest_client.mget(urls)
+
+        # create geodataframe from geojson response
+        df = pd.concat(
+            [gpd.GeoDataFrame.from_features(r.json()) for r in results],
+            ignore_index=True,
+        )
+
+        assert len(df) == count
+
+        fnm = field_name_map.CdcEsriFieldNameMapFactory(geographic_scale, year)
+
+        # map of dataset field names to canonical field names
+        field_names = {
+            v: k
+            for k, v in fnm.dict(exclude_unset=True, exclude={"svi_edition"}).items()
+        }
+
+        df = df.rename(columns=field_names)
+
+        # create missing fields if required
+        df = fnm.create_missing_fields(df)
+
+        df["svi_edition"] = fnm.svi_edition
+
+        # wide to long format
+        rank_col_names = df.columns.str.contains("rank$")
+
+        df = df.melt(
+            id_vars=df.columns[~rank_col_names],
+            value_vars=df.columns[rank_col_names],
+            var_name="rank_theme",
+            value_name="rank",
+        )
+
+        value_col_names = df.columns.str.contains("value$")
+        # some datasources do not include summed theme values
+        if not (value_col_names == False).all():
+            df = df.melt(
+                id_vars=df.columns[~value_col_names],
+                value_vars=df.columns[value_col_names],
+                var_name="value_theme",
+                value_name="value",
+            )
+        # create theme column by truncating rank_theme's _rank suffix
+        df["theme"] = df["rank_theme"].str.rstrip("_rank")
+
+        # drop unnecessary cols
+        # value_theme column might not exist, so ignore errors when trying to drop
+        df = df.drop(columns=["rank_theme", "value_theme"], errors="ignore")
+
+        # lowercase and strip all leading and trailing white spaces from str columns for consistent
+        # output and quality control
+        df_dtypes = df.dtypes
+        str_cols = df_dtypes[df_dtypes == "object"].index
```

Comment:
This might be a dumb question, but does this _always_ select _only_ string columns?

Reply:
I don't think it was a dumb question. I don't think it will _always_ return a string column. I think datetime columns will also be included. Because of this comment, I changed this check to use `pd.DataFrame.select_dtypes` as you showed above.